### PR TITLE
docs: Add "Migrating from Chai" section

### PR DESCRIPTION
**Pull request by caweidmann**
## Proposed changes

Adds new "Migrating from Chai" section to "Assertion" page (https://github.com/webdriverio/webdriverio/issues/5537).

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

### CODE REVIEW

File: docs/Assertion.md [hunk 1]
```diff
@@ -40,3 +40,90 @@ await expect(selectOptions).toHaveChildren({ gte: 1 })
 <!--END_DOCUSAURUS_CODE_TABS-->
 
 For the full list, see the [expect API doc](/docs/api/expect.html).
+
+
+## Migrating from Chai
+
+Chai and expect-webdriverio can coexist, and with some minor adjustments a smooth transition to expect-webdriverio can be achieved. If you've upgraded to wdio v6 then by default you will have access to all the assertions from expect-webdriverio out of the box. This means that globally wherever you use `expect` you would call an expect-webdriverio assertion. That is, unless you have explicitly overriden the global `expect` to use Chai. In this case you would not have access to any of the expect-webdriverio assertions without explicitly importing the expect-webdriverio package where you need it.
```
Comment (christian-bromann) [inline docs/Assertion.md:47]:
```suggestion
[Chai](https://www.chaijs.com/) and [expect-webdriverio](https://github.com/webdriverio/expect-webdriverio#readme) can coexist, and with some minor adjustments a smooth transition to expect-webdriverio can be achieved. If you've upgraded to wdio v6 then by default you will have access to all the assertions from expect-webdriverio out of the box. This means that globally wherever you use `expect` you would call an expect-webdriverio assertion. That is, unless you have explicitly overriden the global `expect` to use Chai. In this case you would not have access to any of the expect-webdriverio assertions without explicitly importing the expect-webdriverio package where you need it.
```

File: docs/Assertion.md [hunk 2]
```diff
@@ -40,3 +40,90 @@ await expect(selectOptions).toHaveChildren({ gte: 1 })
...
+
+```js
+// myfile.js
+const expectChai = require('chai').expect;
+
+describe('Element', function () {
+    it('should be displayed', function (element) {
+        const isDisplayed = browser.$(element).isDisplayed()
+        expectChai(isDisplayed).to.equal(true); // Chai assertion
+    });
```
Comment (christian-bromann) [inline docs/Assertion.md:89]:
```suggestion
    it('should be displayed', () => {
        const isDisplayed = browser.$("#element").isDisplayed()
        expectChai(isDisplayed).to.equal(true); // Chai assertion
    });
```

File: docs/Assertion.md [hunk 3]
```diff
@@ -40,3 +40,90 @@ await expect(selectOptions).toHaveChildren({ gte: 1 })
...
+});
+
+describe('Other element', function () {
+    it('should not be displayed', function (element) {
+        expect(browser.$(element)).not.toBeDisplayed(); // expect-webdriverio assertion
+    });
```
Comment (christian-bromann) [inline docs/Assertion.md:95]:
```suggestion
    it('should not be displayed', () => {
        expect(browser.$("#element")).not.toBeDisplayed(); // expect-webdriverio assertion
    });
```

File: docs/Assertion.md [hunk 4]
```diff
@@ -40,3 +40,90 @@ await expect(selectOptions).toHaveChildren({ gte: 1 })
...
+});
+```
+
+### Global
+
+Assume `expect` was globally overridden to use Chai. In order to use expect-webdriverio assertions we need to globally set a variable in the "before" hook, e.g.:
+
+```js
+// wdio.conf.js
+before: function before() {
```
Comment (christian-bromann) [inline docs/Assertion.md:105]:
```suggestion
before: () => {
```

File: docs/Assertion.md [hunk 5]
```diff
@@ -40,3 +40,90 @@ await expect(selectOptions).toHaveChildren({ gte: 1 })
...
+    const chai = require('chai');
+    global.expect = chai.expect;
+}
+```
+
+Now Chai and expect-webdriverio can be used alongside each other. Once all Chai assertions have been replaced thoughout the code base the "before" hook can be deleted. In your code you would use Chai and expect-webdriverio assertions as follows, e.g.:
+
+```js
+// myfile.js
+describe('Element', function () {
```
Comment (christian-bromann) [inline docs/Assertion.md:117]:
```suggestion
describe('Element', () => {
```

File: docs/Assertion.md [hunk 6]
```diff
@@ -40,3 +40,90 @@ await expect(selectOptions).toHaveChildren({ gte: 1 })
...
+    it('should be displayed', function (element) {
+        const isDisplayed = browser.$(element).isDisplayed()
+        expect(isDisplayed).to.equal(true); // Chai assertion
+    });
```
Comment (christian-bromann) [inline docs/Assertion.md:121]:
```suggestion
    it('should be displayed', () => {
        const isDisplayed = browser.$("#element").isDisplayed()
        expect(isDisplayed).to.equal(true); // Chai assertion
    });
```

File: docs/Assertion.md [hunk 7]
```diff
@@ -40,3 +40,90 @@ await expect(selectOptions).toHaveChildren({ gte: 1 })
...
+});
+
+describe('Other element', function () {
+    it('should not be displayed', function (element) {
+        expectWdio(browser.$(element)).not.toBeDisplayed(); // expect-webdriverio assertion
+    });
+});
```
Comment (christian-bromann) [inline docs/Assertion.md:128]:
```suggestion
describe('Other element', () => {
    it('should not be displayed', () => {
        expectWdio(browser.$("#element")).not.toBeDisplayed(); // expect-webdriverio assertion
    });
});
```

File: docs/Assertion.md (hunk 3 above)
Comment (christian-bromann) [inline docs/Assertion.md:95]:
```suggestion
describe('Other element', () => {
    it('should not be displayed', () => {
        expect(browser.$("#element")).not.toBeDisplayed(); // expect-webdriverio assertion
    });
```

File: docs/Assertion.md (hunk 2 above)
Comment (christian-bromann) [inline docs/Assertion.md:89]:
```suggestion
    it('should be displayed', () => {
        const isDisplayed = browser.$("#element").isDisplayed()
        expectChai(isDisplayed).to.equal(true); // Chai assertion
    });
```

File: docs/Assertion.md [hunk 8]
```diff
@@ -40,3 +40,90 @@ await expect(selectOptions).toHaveChildren({ gte: 1 })
...
+```
+
+To migrate this code remove the Chai import and use the new expect-webdriverio assertion method `toHaveUrl` instead:
+
+```js
+// myfile.js - migrated code
+describe('Homepage', function () {
+    it('should assert', function () {
```
Comment (christian-bromann) [inline docs/Assertion.md:72]:
```suggestion
describe('Homepage', () => {
    it('should assert', () => {
```

File: docs/Assertion.md [hunk 9]
```diff
@@ -40,3 +40,90 @@ await expect(selectOptions).toHaveChildren({ gte: 1 })
...
+### Local
+
+Assume Chai was imported explicitly in a file, e.g.:
+
+```js
+// myfile.js - original code
+const expectChai = require('chai').expect;
+
+describe('Homepage', function () {
+    it('should assert', function () {
```
Comment (christian-bromann) [inline docs/Assertion.md:60]:
```suggestion
describe('Homepage', () => {
    it('should assert', () => {
```

**Comment by caweidmann** [comment]
Haha, thank you!

**Pull request merged**